code/script.js

```diff
@@ -0,0 +1,159 @@
+        drawChart([numberOfProjects, totalProjectsDuringBootcamp - numberOfProjects])
+        getPullRequests(myRepos)
+    });
+
+//Remember to pass along your filtered repos as an argument when
+//you are calling this function
+
+const getPullRequests = (repos) => {
+    //Get all the PRs for each project.
+    repos.forEach(repo => {
+        fetch('https://api.github.com/repos/technigo/' + repo.name + '/pulls' + '?per_page=100', options)
+            .then(res => res.json())
+            .then(data => {
+
+                const repoBranchName = repo.default_branch
+                const repoName = repo.name
+                const repoUrl = repo.html_url
+
+                projects.innerHTML += `
+                <div class="repo-card">
+                    <div class="repo" id=${repoName}>
+                    <h2>
+                        <a href=${repoUrl}> 
+                            ${repoName}:${repoBranchName}
+                        </a>
+                    </h2>
+                    </div>
+                </div>
+                `
+
+
+                const myPullRequests = data.filter((myPR) => myPR.user.login == repo.owner.login);
+                handlePullRequest(repoName, myPullRequests)
+            })
+    })
+
+}
+
+const handlePullRequest = (repoElementId, myPullRequests) => {
+
+    //Get all the PRs for each project.
+    myPullRequests
+        .forEach((pullRequest) => {
+            const prTitle = pullRequest.title
+            const prUrl = pullRequest.html_url
+            const repo = document.getElementById(repoElementId);
+            repo.innerHTML += `
+            <div class="pull-request">
+                <span>PR: <a href=${prUrl}> ${prTitle}</a></span>
+            </div>
+          `
+        })
+    myPullRequests
+        .map(pullRequest => pullRequest.commits_url)
+        .forEach(commitUrl => {
+            fetch(commitUrl + "?per_page=100", options)
+                .then(res => res.json())
+                .then(data => {
+                    const mostRecentCommit = data[data.length - 1]
+                    const commitMessage = mostRecentCommit.commit.message
+                    const commitAuthor = mostRecentCommit.author.login
+                    const commitUrl = mostRecentCommit.html_url
+                    const commitDateString = mostRecentCommit.commit.author.date
+                    const commitDate = Date.parse(commitDateString)
+                    const commitTimeSince = timeSince(commitDate)
+                    const repo = document.getElementById(repoElementId);
+                    repo.innerHTML += `
+                    <div>
+                        <p>${data.length} commits</p>
+                        <p>latest: <a href=${commitUrl}>${commitMessage}</a> by ${commitAuthor} ${commitTimeSince}</p>
+                    </div>
+                `
+                })
+        })
+
+    myPullRequests
+        .map(pullRequest => pullRequest.review_comments_url)
```

Comment:
I just wonder if you need both map and forEach?

Reply:
I separated the 2 just to have a bit cleaner code. It is totally optional :) Thanks for checking though!